drupal/sites/default/boinc/modules/boincwork/includes/boincwork.forms.inc

```diff
@@ -1635,13 +1635,15 @@ function communityprefs_form_submit($form, &$form_state) {
   // Private message settings
   pm_email_notify_user('submit', $edit, $user);
   
-  // Avatar settings
-  if (!$edit['field_image']) $edit['field_image'] = array();
-  $profile_node->field_image = $edit['field_image'];
-  node_save($profile_node);
-  // Flush this from the node cache or changes won't show up immediately!
-  $profile_node = node_load($profile_node->nid, NULL, TRUE);
-  
+  // Avatar settings - only set if profile_node exists.
+  if (!empty($profile_node)) {
```

Comment:
See [PHP type comparison tables](http://php.net/manual/en/types.comparisons.php)
```php
if ($profile_node) {
    $profile_node->field_image = $edit['field_image'] ?: [];
    node_save($profile_node);
    // Flush this from the node cache or changes won't show up immediately!
    $profile_node = node_load($profile_node->nid, NULL, TRUE);
}
```

Reply:
Thanks for your review. The author is currently out of office and will get back to you as soon as he's back.

Reply:
@tristanolive any opinion on this?